File: UniqueDeviceId.js
/**
	Cordova UniqueDeviceId Plugin
	MIT Licensed
**/

if (typeof Cordova !== "undefined") {
    /**
     * Constructor
     */
    function UniqueDeviceId() {
        this._callback = null;
    }

    /**
     * getUniqueDeviceId - use this method when you need a unique identifier in one app
     */
    UniqueDeviceId.prototype.getUniqueDeviceId = function(cb) {
        this._callback = cb;
        Cordova.exec("UniqueDeviceId.getUniqueDeviceId");
    }

    /**
     * uniqueGlobalDeviceIdentifier - use this method when you need a unique global identifier to track a device
     */
    UniqueDeviceId.prototype.getUniqueGlobalDeviceId = function(cb) {
        this._callback = cb;
        Cordova.exec("UniqueDeviceId.getUniqueGlobalDeviceId");
    }

    UniqueDeviceId.prototype._return_callback = function(val) {
        if (this._callback)
            this._callback(val);
    }

    Cordova.addConstructor(function() {
        if(!window.plugins)
        {
            window.plugins = {};
        }
        window.plugins.uniqueDeviceId = new UniqueDeviceId();
    });
};

File: UniqueDeviceId.m
//
//  UniqueDeviceId.m
//	MIT Licensed

#import "UniqueDeviceId.h"

#import "UIDevice+IdentifierAddition.h"

@implementation UniqueDeviceId

- (void)getUniqueDeviceId:(NSMutableArray*)args withDict:(NSMutableDictionary*)opt
{
	NSString *uid = [[UIDevice currentDevice] uniqueDeviceIdentifier];

	NSString* jsCallback = [NSString stringWithFormat:@"window.plugins.uniqueDeviceId._return_callback(\"%@\");", uid];

	[super writeJavascript:jsCallback];
}

- (void)getUniqueGlobalDeviceId:(NSMutableArray*)args withDict:(NSMutableDictionary*)opt
{
	NSString *uid = [[UIDevice currentDevice] uniqueGlobalDeviceIdentifier];

	NSString* jsCallback = [NSString stringWithFormat:@"window.plugins.uniqueDeviceId._return_callback(\"%@\");", uid];

	[super writeJavascript:jsCallback];
}

@end

